framework/src/materials/GenericFunctionVectorMaterial.C

```diff
@@ -0,0 +1,104 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "GenericFunctionVectorMaterial.h"
+#include "Function.h"
+
+registerMooseObject("MooseApp", GenericFunctionVectorMaterial);
+registerMooseObject("MooseApp", ADGenericFunctionVectorMaterial);
+
+defineLegacyParams(GenericFunctionVectorMaterial);
+
+template <bool is_ad>
+InputParameters
+GenericFunctionVectorMaterialTempl<is_ad>::validParams()
+{
+
+  InputParameters params = Material::validParams();
+  params.addClassDescription("Material object for declaring vector properties that are populated "
+                             "by evaluation of Function objects.");
+  params.addParam<std::vector<std::string>>("prop_names",
+                                            "The names of the properties this material will have");
+  params.addParam<std::vector<FunctionName>>("prop_values",
+                                             "The corresponding names of the "
+                                             "functions that are going to provide "
+                                             "the values for the variables, "
+                                             "minor ordering by component");
+  params.addDeprecatedParam<bool>("enable_stateful",
+                                  false,
+                                  "Enable the declaration of old and older values",
+                                  "all properties can implicitly become stateful");
```

Comment:
Probably don't want to add a deprecated parameter in a new object. Recommended to remove and patch the code below not to use it.